### ISSUE TITLE
Support running processes in docker

### ISSUE THREAD

**Issue by domenkozar** [cachix/devenv#2]
Something like `processes.dockerize = true;` would use Docker instead of https://github.com/ddollar/foreman.

Please leave a thumbs up if you'd like this implemented.

**Comment by domenkozar** [comment]
Could you also please add why you'd want to run these in a docker container?

**Comment by domenkozar** [comment]
@arathunku @chenzhekl @ekremparlak @MrSaints @onatm @epk @c4milo @lbodor @Philipp-M @iNode

devenv 0.6 supports [generating containers](https://devenv.sh/containers/), does that support your use case?

**Comment by chenzhekl** [comment]
@domenkozar Yes, it perfectly solves my need, where I have to develop on machines with Docker but without nix installed sometimes. Thank you for your work.

**Comment by domenkozar** [comment]
This probably needs to expose an interface like `docker.processes.XXX.YYY` and do `docker run` in the background. Also needs to think for podman support.
